src/kOS/Function/PrintList.cs

```diff
@@ -189,8 +189,10 @@ private kList GetResourceList(SharedObjects shared)
             foreach (Part part in shared.Vessel.Parts)
             {
                 string stageStr = part.inverseStage.ToString();
-                foreach (PartResource resource in part.Resources)
+                PartResource resource;
+                for (int i = 0; i < part.Resources.Count; ++i)
```

Comment:
Squad removed a lot of enumerators.  Then they put some back in... I didn't go back and find every time that enumerators got re-added however.

Essentially thee idea is that frequent use of enumerators is expensive, because the enumerator needs to allocate a new object when it's used.  For most of our stuff, it isn't a big deal because we don't use them multiple times per tick and because we compile using MS's compiler.  The Mono GC is still a bit of an issue too.  But for resources it's a really big deal for Squad, since they may need to call the parts walk many times per tick (at least once per engine).  Using a traditional for loop only allocates the integer, assuming the list is already constructed.